stage-2/module-pattern/exercise.js

```diff
@@ -17,7 +17,7 @@
  * |          |     |                |   |      |  |             |
  * | protocol |     |    domain      |   | path |  | querystring |
  */
-var UrlParser = (function () {
+var UrlParser = function() {
```

Comment:
I think this was missing an invocation at the end. It should be:

```js
var UrlParser = (function () {
  // ...
})();
```

rather than just a plain function:
```js
var UrlParser = function () {
  // ...
}
```

Otherwise I don't think the tests work. Compare it to the solution.